staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -384,7 +384,9 @@ func NewCacherFromConfig(config Config) (*Cacher, error) {
 			}, time.Second, stopCh,
 		)
 	}()
-
+	
+	objectType := objType.String()
+	watchCacheCapacity.WithLabelValues(objectType).Set(float64(cacher.watchCache.capacity))
```

Comment:
I'm supportive for adding this metric, but it's not the right place to set it.
Especially, given that the watchcache size is now dynamic.

This should basically be set:
- on watch cache creation
- whenever its size changes:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go#L382

Reply:
yeah, there are two place to set:

> on watch cache creation:

watchCache is created on https://github.com/yuanhuaiwang/kubernetes/blob/67b78a8cee0fe6c43afb1ed87d6e3664af78070d/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L361, so metric init should be add in here? or add metric here on https://github.com/yuanhuaiwang/kubernetes/blob/67b78a8cee0fe6c43afb1ed87d6e3664af78070d/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go#L71

> whenever its size changes:

func recordsWatchCacheCapacityChange record watchCache capacity resize(increase or decrease) operations: https://github.com/yuanhuaiwang/kubernetes/blob/67b78a8cee0fe6c43afb1ed87d6e3664af78070d/staging/src/k8s.io/apiserver/pkg/storage/cacher/metrics.go#L77, so i think  it is right here, what do you think?

Reply:
@wojtek-t please take a look!

Reply:
OK - the later looks fine.

But please move this one to newWatchCache function

Reply:
I have moved this to newWatchCache function, can you  add label ok-to-test and approve it ? please tell me if have something , thanks a lot ! @wojtek-t